core/src/main/scala/org/apache/spark/SparkStatusTracker.scala

```diff
@@ -104,4 +106,34 @@ class SparkStatusTracker private[spark] (sc: SparkContext) {
       }
     }
   }
+
+  /**
+   * Returns a list of all known executors, represented by string with format: "host:port"
+   */
+  def getExecutors(): Array[String] = {
```

Comment:
The other thing is I don't know if we want to query the scheduler every time we want a list of executors.